### ISSUE TITLE
master/branch-3.2: GCE backend is broken

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla-cluster-tests#1472]
### Prerequisites

* [x] Are you rebased to master ?
* [x] Is it reproducible ?
* [x] Did you perform a cursory search if this issue isn't opened ?

### Versions
* **SCT**: master & branch-3.2

### Logs
* test job:
    - https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-centos7/6
    - https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-centos7/8

### Steps to Reproduce

1. run upgrade test by https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-centos7

**Expected behavior:** [What you expected to happen]

PASS

**Actual behavior:** [What actually happened]

Job failed in prepare stage:

```
> Traceback (most recent call last):
>   File "/sct/sdcm/tester.py", line 134, in wrapper
>     return method(*args, **kwargs)
>   File "/sct/sdcm/utils/common.py", line 136, in inner
>     res = func(*args, **kwargs)
>   File "/sct/sdcm/tester.py", line 259, in setUp
>     self.init_resources()
>   File "/sct/sdcm/tester.py", line 808, in init_resources
>     monitor_info=monitor_info)
>   File "/sct/sdcm/tester.py", line 482, in get_cluster_gce
>     **common_params)
>   File "/sct/sdcm/cluster_gce.py", line 380, in __init__
>     params=params)
>   File "/sct/sdcm/cluster_gce.py", line 181, in __init__
>     region_names=gce_region_names)
>   File "/sct/sdcm/cluster.py", line 2346, in __init__
>     self.add_nodes(n_nodes)
>   File "/sct/sdcm/cluster_gce.py", line 310, in add_nodes
>     instances = self._create_instances(count, dc_idx)
>   File "/sct/sdcm/cluster_gce.py", line 277, in _create_instances
>     instances.append(self._create_instance(node_index, dc_idx))
>   File "/sct/sdcm/cluster_gce.py", line 249, in _create_instance
>     dc_idx=dc_idx))
>   File "/sct/sdcm/cluster_gce.py", line 208, in _get_root_disk_struct
>     "diskType": self._get_disk_url(disk_type, dc_idx=dc_idx),
>   File "/sct/sdcm/cluster_gce.py", line 199, in _get_disk_url
>     return "projects/%s/zones/%s/diskTypes/%s" % (project.name, self._gce_region_names[dc_idx], disk_type)
> TypeError: 'NoneType' object is not subscriptable
```

```
> Uncaught exception < t:2019-11-25 04:42:29,657 f:log.py          l:20   c:sdcm.utils.log       p:ERROR > Uncaught exception
> Traceback (most recent call last):
>   File "./sct.py", line 486, in <module>
>     cli()
>   File "/usr/local/lib64/python3.6/site-packages/click/core.py", line 764, in __call__
>     return self.main(*args, **kwargs)
>   File "/usr/local/lib64/python3.6/site-packages/click/core.py", line 717, in main
>     rv = self.invoke(ctx)
>   File "/usr/local/lib64/python3.6/site-packages/click/core.py", line 1137, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "/usr/local/lib64/python3.6/site-packages/click/core.py", line 956, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/usr/local/lib64/python3.6/site-packages/click/core.py", line 555, in invoke
>     return callback(*args, **kwargs)
>   File "./sct.py", line 413, in run_test
>     failfast=False, buffer=False, catchbreak=True)
>   File "/usr/lib64/python3.6/unittest/main.py", line 94, in __init__
>     self.parseArgs(argv)
>   File "/usr/lib64/python3.6/unittest/main.py", line 141, in parseArgs
>     self.createTests()
>   File "/usr/lib64/python3.6/unittest/main.py", line 148, in createTests
>     self.module)
>   File "/usr/lib64/python3.6/unittest/loader.py", line 219, in loadTestsFromNames
>     suites = [self.loadTestsFromName(name, module) for name in names]
>   File "/usr/lib64/python3.6/unittest/loader.py", line 219, in <listcomp>
>     suites = [self.loadTestsFromName(name, module) for name in names]
>   File "/usr/lib64/python3.6/unittest/loader.py", line 197, in loadTestsFromName
>     inst = parent(name)
>   File "/sct/sdcm/tester.py", line 149, in __init__
>     self.params.verify_configuration()
>   File "/sct/sdcm/sct_config.py", line 1155, in verify_configuration
>     _check_backend_defaults(backend, self.backend_required_params[backend])
>   File "/sct/sdcm/sct_config.py", line 1151, in _check_backend_defaults
>     assert _opt['name'] in self, "{} missing from config for {}".format(_opt['name'], backend)
> AssertionError: scylla_repo missing from config for gce
```

/CC @fruch @bentsi @roydahan

**Comment by amoskong** [comment]
This this issue by the second patch in https://github.com/scylladb/scylla-cluster-tests/pull/1470/commits 
- https://github.com/scylladb/scylla-cluster-tests/pull/1470/commits/e485426de1c01fdbc61b365de8c29f86af9287d0

**Comment by amoskong** [comment]
network isn't available during scylla_setup, only occurred once:
Let's see if it occurs again on 3.2 upgrade jobs.

```
> Traceback (most recent call last):
>   File "/sct/sdcm/cluster.py", line 2489, in node_setup
>     cl_inst.node_setup(node, **setup_kwargs)
>   File "/sct/sdcm/cluster.py", line 2984, in node_setup
>     node.remoter.run('sudo cat /etc/scylla.d/io.conf')
>   File "/sct/sdcm/remote.py", line 239, in run
>     result = _run()
>   File "/sct/sdcm/utils/common.py", line 89, in inner
>     return func(*args, **kwargs)
>   File "/sct/sdcm/remote.py", line 230, in _run
>     watchers=watchers, timeout=timeout)
>   File "<decorator-gen-3>", line 2, in run
>   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 30, in opens
>     return method(self, *args, **kwargs)
>   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 721, in run
>     return self._run(self._remote_runner(), command, **kwargs)
>   File "/usr/local/lib/python3.6/site-packages/invoke/context.py", line 101, in _run
>     return runner.run(command, **kwargs)
>   File "/usr/local/lib/python3.6/site-packages/invoke/runners.py", line 291, in run
>     return self._run_body(command, **kwargs)
>   File "/usr/local/lib/python3.6/site-packages/invoke/runners.py", line 315, in _run_body
>     self.start(command, shell, env)
>   File "/usr/local/lib/python3.6/site-packages/fabric/runners.py", line 36, in start
>     self.channel = self.context.create_session()
>   File "<decorator-gen-2>", line 2, in create_session
>   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 30, in opens
>     return method(self, *args, **kwargs)
>   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 698, in create_session
>     channel = self.transport.open_session()
>   File "/usr/local/lib/python3.6/site-packages/paramiko/transport.py", line 879, in open_session
>     timeout=timeout,
>   File "/usr/local/lib/python3.6/site-packages/paramiko/transport.py", line 1006, in open_channel
>     raise e
>   File "/usr/local/lib/python3.6/site-packages/paramiko/transport.py", line 2055, in run
>     ptype, m = self.packetizer.read_message()
>   File "/usr/local/lib/python3.6/site-packages/paramiko/packet.py", line 459, in read_message
>     header = self.read_all(self.__block_size_in, check_rekey=True)
>   File "/usr/local/lib/python3.6/site-packages/paramiko/packet.py", line 301, in read_all
>     x = self.__socket.recv(n)
> ConnectionResetError: [Errno 104] Connection reset by peer
```

**Comment by amoskong** [comment]
> network isn't available during scylla_setup, only occurred once:
> Let's see if it occurs again on 3.2 upgrade jobs.
> 
> ```
> > Traceback (most recent call last):
> >   File "/sct/sdcm/cluster.py", line 2489, in node_setup
> >     cl_inst.node_setup(node, **setup_kwargs)
> >   File "/sct/sdcm/cluster.py", line 2984, in node_setup
> >     node.remoter.run('sudo cat /etc/scylla.d/io.conf')
> >   File "/sct/sdcm/remote.py", line 239, in run
> >     result = _run()
> >   File "/sct/sdcm/utils/common.py", line 89, in inner
> >     return func(*args, **kwargs)
> >   File "/sct/sdcm/remote.py", line 230, in _run
> >     watchers=watchers, timeout=timeout)
> >   File "<decorator-gen-3>", line 2, in run
> >   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 30, in opens
> >     return method(self, *args, **kwargs)
> >   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 721, in run
> >     return self._run(self._remote_runner(), command, **kwargs)
> >   File "/usr/local/lib/python3.6/site-packages/invoke/context.py", line 101, in _run
> >     return runner.run(command, **kwargs)
> >   File "/usr/local/lib/python3.6/site-packages/invoke/runners.py", line 291, in run
> >     return self._run_body(command, **kwargs)
> >   File "/usr/local/lib/python3.6/site-packages/invoke/runners.py", line 315, in _run_body
> >     self.start(command, shell, env)
> >   File "/usr/local/lib/python3.6/site-packages/fabric/runners.py", line 36, in start
> >     self.channel = self.context.create_session()
> >   File "<decorator-gen-2>", line 2, in create_session
> >   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 30, in opens
> >     return method(self, *args, **kwargs)
> >   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 698, in create_session
> >     channel = self.transport.open_session()
> >   File "/usr/local/lib/python3.6/site-packages/paramiko/transport.py", line 879, in open_session
> >     timeout=timeout,
> >   File "/usr/local/lib/python3.6/site-packages/paramiko/transport.py", line 1006, in open_channel
> >     raise e
> >   File "/usr/local/lib/python3.6/site-packages/paramiko/transport.py", line 2055, in run
> >     ptype, m = self.packetizer.read_message()
> >   File "/usr/local/lib/python3.6/site-packages/paramiko/packet.py", line 459, in read_message
> >     header = self.read_all(self.__block_size_in, check_rekey=True)
> >   File "/usr/local/lib/python3.6/site-packages/paramiko/packet.py", line 301, in read_all
> >     x = self.__socket.recv(n)
> > ConnectionResetError: [Errno 104] Connection reset by peer
> ```

This problem occurred again on
- https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-centos7/9/console
- https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-debian9/7/console
- https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-ubuntu16.04/4/console
- https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-ubuntu18.04/3/console

Hi @aleksbykov , do we change the code of remoter / ssh connection recently?

**Comment by amoskong** [comment]
I created a gce instance, and installed scylla manually.
I didn't touch ssh connection stuck, or system stcuk after scylla_setup:
- sudo /usr/lib/scylla/scylla_setup --nic eth0 --disks /dev/nvme0n1,/dev/nvme0n2
- sudo sync

I also didn't touch this problem when I test upgrade of 3.2 by SCT branch-3.1

**Comment by amoskong** [comment]
The upgrade test runs well with SCT branch-3.1:
- https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-centos7/12/console (RUNNING ...)

So it might the problem of SCT.

**Comment by bentsi** [comment]
Not related to SCT.

How did we use 70TB in one region?
QuotaExceededError: u"Quota 'LOCAL_SSD_TOTAL_GB' exceeded.  Limit: 70000.0 in region us-east1."

**Comment by bentsi** [comment]
@amoskong I don't know how you got to the conclusion that GCE backend is broken. Each one of the runs that I looked at has a different issue. Please try to be more specific. Let's close this and open a specific ones.

**Comment by amoskong** [comment]
> Not related to SCT.
> 
> How did we use 70TB in one region?
> Job 11: https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-centos7/11/console

each upgrade test has 4 db nodes, each db node has 3 local ssd (375G).
so each upgrade job uses 4 * 3 * 375G = 4.5T

There are also some big disks used by instances that were created manually.

> ```
> QuotaExceededError: u"Quota 'LOCAL_SSD_TOTAL_GB' exceeded.  Limit: 70000.0 in region us-east1."
> ```

 
> Latest job failed due to Scylla issue: https://jenkins.scylladb.com/view/scylla-3.2/job/scylla-3.2/job/rolling-upgrade/job/rolling-upgrade-centos7/12/console

This job doesn't use SCT branch-3.2, but a private branch based on branch-3.1

**Comment by bentsi** [comment]
@amoskong I think I found the issue. Working on PR

**Comment by amoskong** [comment]
> @amoskong I think I found the issue. Working on PR

Which issue?

**Comment by amoskong** [comment]
> @amoskong I think I found the issue. Working on PR

Current major concern is the ConnectionResetError: https://github.com/scylladb/scylla-cluster-tests/issues/1472#issuecomment-558005428

This PR https://github.com/scylladb/scylla-cluster-tests/pull/1470 contained a patch to fix the following issue:

> Traceback (most recent call last):
>   File "/sct/sdcm/tester.py", line 134, in wrapper
>     return method(*args, **kwargs)
>   File "/sct/sdcm/utils/common.py", line 136, in inner
>     res = func(*args, **kwargs)
>   File "/sct/sdcm/tester.py", line 259, in setUp
>     self.init_resources()
>   File "/sct/sdcm/tester.py", line 808, in init_resources
>     monitor_info=monitor_info)
>   File "/sct/sdcm/tester.py", line 482, in get_cluster_gce
>     **common_params)
>   File "/sct/sdcm/cluster_gce.py", line 380, in __init__
>     params=params)
>   File "/sct/sdcm/cluster_gce.py", line 181, in __init__
>     region_names=gce_region_names)
>   File "/sct/sdcm/cluster.py", line 2346, in __init__
>     self.add_nodes(n_nodes)
>   File "/sct/sdcm/cluster_gce.py", line 310, in add_nodes
>     instances = self._create_instances(count, dc_idx)
>   File "/sct/sdcm/cluster_gce.py", line 277, in _create_instances
>     instances.append(self._create_instance(node_index, dc_idx))
>   File "/sct/sdcm/cluster_gce.py", line 249, in _create_instance
>     dc_idx=dc_idx))
>   File "/sct/sdcm/cluster_gce.py", line 208, in _get_root_disk_struct
>     "diskType": self._get_disk_url(disk_type, dc_idx=dc_idx),
>   File "/sct/sdcm/cluster_gce.py", line 199, in _get_disk_url
>     return "projects/%s/zones/%s/diskTypes/%s" % (project.name, self._gce_region_names[dc_idx], disk_type)
> TypeError: 'NoneType' object is not subscriptable

**Comment by bentsi** [comment]
The issue with SSH, meanwhile merged #1470

**Comment by dkropachev** [comment]
Not a valid anymore
